src/modules/PortAudio/PortAudioWriter.cpp

```diff
@@ -203,12 +240,36 @@ bool PortAudioWriter::open()
 
 bool PortAudioWriter::openStream()
 {
+#ifdef Q_OS_MACOS
+	bool bitPerfectMode = sets().getBool("BitPerfect");
+	if (bitPerfectMode && (outputDevice.isEmpty() || outputDevice == "Default"))
+	{
+		int chn = getParam("chn").toInt();
+		outputParameters.device = PortAudioCommon::getDeviceIndexForOutput(outputDevice, chn);
+	}
+#endif
 	PaStream *newStream = nullptr;
 	if (Pa_OpenStream(&newStream, nullptr, &outputParameters, sample_rate, 0, paDitherOff, nullptr, nullptr) == paNoError)
 	{
 		stream = newStream;
 		outputLatency = Pa_GetStreamInfo(stream)->outputLatency;
 		modParam("delay", outputLatency);
+#ifdef Q_OS_MACOS
+		modParam("rate", Pa_GetStreamInfo(stream)->sampleRate);
+		if (bitPerfectMode)
+		{
+			const QString devName = QString::fromUtf8(Pa_GetDeviceInfo(outputParameters.device)->name);
```

Comment:
`QString::fromUtf8` does nothing, you can remove this.

Reply:
I'm used to KDE where the ctors taking C strings have been disabled...